File: backed/src/main/java/com/knowledge/entity/CustomPractice.java
package com.knowledge.entity;

import com.knowledge.common.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 专项实战实体类
 * 针对特定考试、项目、竞赛的定制化训练
 *
 * @author system
 * @date 2025-01-15
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class CustomPractice extends BaseEntity {
    
    /**
     * 实战名称
     */
    private String practiceName;
    
    /**
     * 场景类型（考试/项目/竞赛）
     */
    private String scenarioType;
    
    /**
     * 场景详情（具体考试名称、项目类型、竞赛名称等）
     */
    private String scenarioDetail;
    
    /**
     * 难度等级（简单/中等/困难）
     */
    private String difficulty;
    
    /**
     * 背景说明（考试大纲、项目背景、竞赛规则等）
     */
    private String background;
    
    /**
     * 详细描述
     */
    private String description;
    
    /**
     * 目标技能点（JSON格式）
     */
    private String targetSkills;
    
    /**
     * 指定技术栈（JSON格式）
     */
    private String techStack;
    
    /**
     * 限制条件（JSON格式：时间、资源、技术限制等）
     */
    private String constraints;
    
    /**
     * 实战内容（JSON格式）
     */
    private String practiceContent;
    
    /**
     * 评估标准（JSON格式）
     */
    private String evaluationCriteria;
    
    /**
     * 参考资料（JSON格式）
     */
    private String referenceMaterials;
    
    /**
     * 建议完成时间（分钟）
     */
    private Integer timeLimit;
    
    /**
     * 模板兼容性说明
     */
    private String compatibilityNotes;
    
    /**
     * 标签（逗号分隔）
     */
    private String tags;
}

File: fronted/src/api/customPractice.ts
import request from './request'

export interface CustomPractice {
  id?: number
  practiceName: string
  scenarioType: string  // 场景类型（考试/项目/竞赛）
  scenarioDetail?: string  // 场景详情
  difficulty?: string
  background?: string  // 背景说明
  description: string
  targetSkills: string  // JSON string
  techStack?: string  // JSON string - 指定技术栈
  constraints?: string  // JSON string - 限制条件
  practiceContent: string  // JSON string
  evaluationCriteria: string  // JSON string
  referenceMaterials?: string  // JSON string - 参考资料
  timeLimit?: number
  compatibilityNotes?: string  // 模板兼容性说明
  tags?: string
  recCreator?: string
  recCreateTime?: string
  recRevisor?: string
  recReviseTime?: string
}

/**
 * 获取所有定制模拟实战
 */
export const getAllCustomPractices = () => {
  return request.get('/api/custom-practice/list')
}

/**
 * 根据ID获取定制模拟实战
 */
export const getCustomPracticeById = (id: number) => {
  return request.get(`/api/custom-practice/${id}`)
}

/**
 * 创建定制模拟实战
 */
export const createCustomPractice = (data: CustomPractice) => {
  return request.post('/api/custom-practice', data)
}

/**
 * 更新定制模拟实战
 */
export const updateCustomPractice = (data: CustomPractice) => {
  return request.put('/api/custom-practice', data)
}

/**
 * 删除定制模拟实战
 */
export const deleteCustomPractice = (id: number) => {
  return request.delete(`/api/custom-practice/${id}`)
}

